v2/cmd/asoctl/pkg/importresources/resource_importer.go

```diff
@@ -282,3 +291,12 @@ func (ri *ResourceImporter) importResource(
 
 	return result
 }
+
+// desiredWorkers returns the number of workers to use for importing resources.
+func (ri *ResourceImporter) desiredWorkers() int {
```

Comment:
minor: This method isn't hurting but my understanding is that it's not actually possible to take this path unless the user explicitly passes a negative or nonzero value for the cmdline option, because if the user doesn't specify anything the default of 4 is used at the cmdline flag level (thus not hitting the default of 4 here).

I'm wondering if it's better to add a `validate` helper to `importAzureResourceOptions`, and then call that at the start of `importAzureResource` and fail the cmd if the user passes a negative value? Could also be extended to check the provided labels for validity (length, allowed number of characters?).

Oh wait I see you already parse the labels it's just down below after the resources are imported. Maybe move the parsing of the labels up front, and validate input of workers too to fail fast on bad user input, then run import (and use the actual parsed labels if configured?)

Reply:
The default will always be `4` if used from the command-line, but that's not the only way to get here. 
With the relocation under `pkg` we've allowed for reuse. The value here is an option provided in the `ResourceImporterOptions` struct, which might not be set by a caller.